pennylane/devices/qubit/adjoint_jacobian.py

```diff
@@ -231,15 +329,21 @@ def adjoint_vjp(tape: QuantumTape, cotangents: Tuple[Number], state=None):
 
         The adjoint differentiation method has the following restrictions:
 
-        * Only expectation values are supported as measurements.
-
         * Cannot differentiate with respect to observables.
 
         * Observable being measured must have a matrix.
 
     Args:
         tape (QuantumTape): circuit that the function takes the gradient of
-        cotangents (Tuple[Number]): gradient vector for output parameters
+        cotangents (Tuple[Number]): gradient vector for output parameters. For computing
+            the full Jacobian, the cotangents can be batched to vectorize the computation.
+            In this case, the cotangents can have the following shapes. ``batch_size``
+            below refers to the number of entries in the Jacobian:
```

Comment:
and the number of entries in the jacobian refers to a parameter batch dimension? or the number of parameters? tests suggest that it's only about a batch dimension for the cotangents themselves, in which case, I'm wondering what causes this cotangent batch dimension?

Reply:
This is "final" jacobian, as in output of the full workflow.

Reply:
Number of entries is the number of total outputs. This can be different from the number of values returned by the QNode as one could use returned expectation values to generate a different number of hybrid outputs.